reactor-netty-core/src/main/java/reactor/netty/transport/TransportConnector.java

```diff
@@ -581,8 +581,10 @@ public ChannelPromise syncUninterruptibly() {
 		}
 
 		@Override
+		@SuppressWarnings("FutureReturnValueIgnored")
 		public boolean tryFailure(Throwable cause) {
 			if (RESULT_UPDATER.compareAndSet(this, null, cause)) {
+				channel.close();
```

Comment:
As this is a bugfix, I don't see a reason why one should disable this.
I also checked all code paths that call `tryFailure` and all of them close the channel already except the bind call.
Also duplicate calls to close do not do any harm as they are guarded in netty and only the first one actually does something.

Reply:
If you don't like this at this place, as an alternative we could move the `channel.close()` only to the `TransportConnector.bind` method. Imho it would be a safer to do it here as it avoid the same problem also for other code paths that fail during channel initialization.

Reply:
We can do it without flag but you need to handle it a bit different. See https://github.com/reactor/reactor-netty/blob/main/reactor-netty-core/src/main/java/reactor/netty/transport/TransportConnector.java#L297-L309